src/main/java/com/google/step/snippet/external/W3SchoolsClient.java

```diff
@@ -49,9 +51,18 @@ public Card search(String w3Link) {
     if (snippets.isEmpty() || snippets.first().getElementsByClass(CODE_CLASS).text().isEmpty()) {
       return null;
     }
-    String title = titles.first().text();
-    String description = descriptions.first().text();
-    String code = snippets.first().getElementsByClass(CODE_CLASS).text();
+    String title =
+        StringEscapeUtils.escapeHtml4(
+            Jsoup.clean(titles.first().text(), Whitelist.basicWithImages()));
+    String description =
+        StringEscapeUtils.escapeHtml4(
+            Jsoup.clean(descriptions.first().text(), Whitelist.basicWithImages()));
+    String code =
+        Jsoup.clean(
+            snippets.first().getElementsByClass(CODE_CLASS).text(), Whitelist.basicWithImages());
+    if (query.contains("html")) {
+      code = StringEscapeUtils.escapeHtml4(code);
```

Comment:
I think at least for W3Schools, checking if the URL also contains html is a more reliable check way to check if escaping is needed. Up to you if you want to do the same thing for GeeksForGeeks

Reply:
ex: just searching "images" yields unescaped html code from W3

Reply:
Yeah, was thinking of other filters we might want to add. There are other keywords (eg. svg, icons, etc.) that would also return html code that isn't escaped.